source/common/upstream/load_balancer_impl.h

```diff
@@ -363,12 +363,17 @@ class RoundRobinLoadBalancer : public EdfLoadBalancerBase {
  */
 class LeastRequestLoadBalancer : public EdfLoadBalancerBase {
 public:
-  LeastRequestLoadBalancer(const PrioritySet& priority_set, const PrioritySet* local_priority_set,
-                           ClusterStats& stats, Runtime::Loader& runtime,
-                           Runtime::RandomGenerator& random,
-                           const envoy::api::v2::Cluster::CommonLbConfig& common_config)
+  LeastRequestLoadBalancer(
+      const PrioritySet& priority_set, const PrioritySet* local_priority_set, ClusterStats& stats,
+      Runtime::Loader& runtime, Runtime::RandomGenerator& random,
+      const envoy::api::v2::Cluster::CommonLbConfig& common_config,
+      const absl::optional<envoy::api::v2::Cluster::LeastRequestLbConfig> least_request_config)
       : EdfLoadBalancerBase(priority_set, local_priority_set, stats, runtime, random,
-                            common_config) {
+                            common_config),
+        least_request_config_(least_request_config),
+        choice_count_(least_request_config_.has_value()
```

Comment:
nit: You can use PROTOBUF_GET_WRAPPED_OR_DEFAULT here.

Reply:
It's optional, so I'd still have to check if `least_request_config_.has_value()`. I can wrap this all in a macro, but I'm not sure it'll look much cleaner.

Reply:
I can't use that macro for the optional type without making it more complicated. It's a good one though- I'll use it moving forward!

Reply:
In looking at this again, I'm not sure if this is correct. Do you need to make the choice_count field both required and >= 2? Or does >= 2 make it required? I'm not sure that it does.

Reply:
It's not required, but if it is populated the proto validation will ensure that it's >=2:
```
  message LeastRequestLbConfig {
    // The number of random healthy hosts from which the host with the fewest active requests will
    // be chosen. Defaults to 2 so that we perform two-choice selection if the field is not set.
    google.protobuf.UInt32Value choice_count = 1 [(validate.rules).uint32.gte = 2];
  }
```

If it's never used, then `least_request_config_` shouldn't have a value and we default it to 2.

Reply:
Ah, I understand now. You're right, I missed that. I'll fix it.